cmake/Modules/FindMKL.cmake

```diff
@@ -41,10 +41,11 @@ IF (WIN32)
 ELSE (WIN32)
   SET(DEFAULT_INTEL_COMPILER_DIR "/opt/intel")
   SET(DEFAULT_INTEL_MKL_DIR "/opt/intel/mkl")
-  if (EXISTS "/opt/intel/oneapi")
-    SET(DEFAULT_INTEL_COMPILER_DIR "/opt/intel/oneapi")
-    if (EXISTS "/opt/intel/oneapi/mkl/latest")
-      SET(DEFAULT_INTEL_MKL_DIR "/opt/intel/oneapi/mkl/latest")
+  SET(DEFAULT_INTEL_ONEAPI_DIR "/opt/intel/oneapi")
+  if (EXISTS "${DEFAULT_INTEL_ONEAPI_DIR}")
+    SET(DEFAULT_INTEL_COMPILER_DIR "${DEFAULT_INTEL_ONEAPI_DIR}")
+    if (EXISTS "${DEFAULT_INTEL_ONEAPI_DIR}/mkl/latest")
+      SET(DEFAULT_INTEL_MKL_DIR "${DEFAULT_INTEL_ONEAPI_DIR}/mkl/latest")
```

Comment:
Adding some explanation to PR description what this change is doing would be nice, imo.

Reply:
I split the large commits into smaller ones and added comments.